lizaku/assignment3/assignment3.py

```diff
@@ -0,0 +1,127 @@
+class Dictionary:
+    '''
+    I implement the dictionary as a trie. As an input I give an array of words 
+    that are then transformed into a trie. isWord and isPrefix are methods of the Dictionary.
+    '''
+    def __init__(self, data):
+        self.data = self._make(data)
+    
+    def _make(self, words):
+        # constructing the trie, it's in form of a dictionary (not very efficient though...)
+        _end = '_end_'
+        root = dict()
+        for word in words:
+            current_dict = root
+            for letter in word:
+                current_dict = current_dict.setdefault(letter, {})
+            current_dict[_end] = _end
+        return root
+    
+    def isWord(self, word):
+        _end = '_end_'
+        current_dict = self.data
+        for letter in word:
+            if letter in current_dict:
+                current_dict = current_dict[letter]
+            else:
+                return False
+        else:
+            if _end in current_dict:
+                return True
+            else:
+                return False
+    
+    def isPrefix(self, word):
+        current_dict = self.data
+        for letter in word:
+            if letter in current_dict:
+                current_dict = current_dict[letter]
+            else:
+                return False
+        return True
+
+
+class Grid:
+    '''
+    Grid is initially a 2D array, but I turn all values into Node objects
+    and augment them with their neighbors
+    '''
+    def __init__(self, data):
+        self.data = self._make(data)
+    
+    def _make(self, data):
```

Comment:
Why did you separate this in a different method? Couldn't it be directly in the init()?

Reply:
But I moved this method to init(), if you think that is more accurate.